### PR TITLE
Default unrefined AdtDef

### LINKED ISSUE #83: Default specs with Polymorphism
Code like the below should work, using just the plain rust type signatures and nothing else

```rust
#![feature(register_tool)]
#![register_tool(lr)]

#[lr::sig(fn() -> i32[1])]
pub fn test2() -> i32 {
   let opt = Option::Some(1);
   opt.unwrap()
}
```

### PR DESCRIPTION
**Pull request by ranjitjhala**
Fixes #83

### CODE REVIEW

File: liquid-rust-tests/tests/neg/structs/default-option.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#![feature(register_tool)]
+#![register_tool(lr)]
+
+// We need this right now to not deal with the enum `Some`
+
+#[lr::assume]
+#[lr::sig(fn(T) -> Option<T>)]
```
Comment (nilehmann) [inline liquid-rust-tests/tests/neg/structs/default-option.rs:7]: You should be able to drop the `lr::sig` as this is the default signature, no?

File: liquid-rust-typeck/src/global_env.rs [hunk 2]
```diff
@@ -24,6 +24,10 @@ pub struct GlobalEnv<'tcx> {
     adt_defs: FxHashMap<DefId, ty::AdtDef>,
 }
 
+fn default_adt_def() -> ty::AdtDef {
+    ty::AdtDef::Opaque { refined_by: crate::intern::Interned::from(vec![]) }
```
Comment (nilehmann) [inline liquid-rust-typeck/src/global_env.rs:28]:
```suggestion
    ty::AdtDef::Opaque { refined_by: crate::intern::List::from(vec![]) }
```

**Review by nilehmann — APPROVED** [review]
@ranjitjhala Nice! Some small comments, but it looks good to me

File: liquid-rust-tests/tests/neg/structs/default-option.rs (hunk 1 above)
Comment (ranjitjhala) [reply liquid-rust-tests/tests/neg/structs/default-option.rs]: Yup, good catch!

**Pull request merged**